oligo_designer_toolsuite/oligo_specificity_filter/_filter_exact_matches.py

```diff
@@ -0,0 +1,203 @@
+import os
+import time
+
+import iteration_utilities
+import pandas as pd
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from joblib import Parallel, delayed, parallel_backend
+
+from oligo_designer_toolsuite.IO._data_parser import read_oligos_DB_tsv
+
+from ._filter_base import ProbeFilterBase
+
+
+class ProbeFilterExact(ProbeFilterBase):
+    def __init__(self, n_jobs, dir_output, dir_annotations):
```

Comment:
this whole class should be rewritten such that we do not operate with reading / writing files but instead directly on the probe DB -> we can call this line ` duplicated_sequences = list(iteration_utilities.unique_everseen(iteration_utilities.duplicates(sequences)))` directly on all sequences of the probe DB and then filter out the returned sequences.